install-from-github.js

```diff
@@ -32,55 +32,31 @@ const fs = require('fs');
 const util = require('util');
 const rmAsync = util.promisify(require('rimraf'));
 const existsAsync = path => fs.promises.access(path).then(() => true, e => false);
-const {downloadBrowserWithProgressBar} = require('./download-browser');
+const {downloadBrowserWithProgressBar, localDownloadOptions} = require('./download-browser');
 const protocolGenerator = require('./utils/protocol-types-generator');
-const packageJSON = require('./package.json');
-
-const DOWNLOADED_BROWSERS_JSON_PATH = path.join(__dirname, '.downloaded-browsers.json');
-const DOWNLOAD_PATHS = {
-  chromium: path.join(__dirname, '.local-browsers', `chromium-${packageJSON.playwright.chromium_revision}`),
-  firefox: path.join(__dirname, '.local-browsers', `firefox-${packageJSON.playwright.firefox_revision}`),
-  webkit: path.join(__dirname, '.local-browsers', `webkit-${packageJSON.playwright.webkit_revision}`),
-};
 
 (async function() {
-  const downloadedBrowsersJSON = await fs.promises.readFile(DOWNLOADED_BROWSERS_JSON_PATH, 'utf8').then(json => JSON.parse(json)).catch(() => ({}));
-  try {
-    if (!(await existsAsync(DOWNLOAD_PATHS.chromium))) {
-      const crExecutablePath = await downloadBrowserWithProgressBar(DOWNLOAD_PATHS.chromium, 'chromium', false /* respectGlobalInstall */);
-      downloadedBrowsersJSON.crExecutablePath = crExecutablePath;
-      await protocolGenerator.generateChromiumProtocol(crExecutablePath);
-      await fs.promises.writeFile(DOWNLOADED_BROWSERS_JSON_PATH, JSON.stringify(downloadedBrowsersJSON));
-    }
-  } catch (e) {
-    console.warn(e.message);
+  const chromiumOptions = localDownloadOptions('chromium');
+  const firefoxOptions = localDownloadOptions('firefox');
+  const webkitOptions = localDownloadOptions('webkit');
+  if (!(await existsAsync(chromiumOptions.downloadPath))) {
+    await downloadBrowserWithProgressBar(chromiumOptions);
+    await protocolGenerator.generateChromiumProtocol(chromiumOptions.executablePath).catch(console.warn);
```

Comment:
Why these catches?

Reply:
- we always install all three browsers on our CI and generate protocols for them
- the `firefox-linux` github action doesn't install chromium dependencies. I'd like to keep it this way so that our github action is lean and can be easily reused
- the `generateChromiumProtocol` fails if there are no chromium dependencies installed

PS: these checks were before in a form of `try-catch`.